Still "Too many connections"

```
Caused by: java.sql.SQLNonTransientConnectionException: Data source rejected establishment of connection,  message from server: "Too many connections"
```

less frequent now. maybe just one or two calls where closing is missing. Maybe auth filter

--- a/src/main/java/io/github/janmalch/kino/control/ManagingControl.java
+++ b/src/main/java/io/github/janmalch/kino/control/ManagingControl.java
@@ -17,9 +17,11 @@ public abstract class ManagingControl<P> implements Control<P>, Manageable {
 
   @Override
   public final <T> T execute(ResultBuilder<T, P> result) {
-    var output = compute(result);
-    this.close();
-    return output;
+    try {
+      return compute(result);
+    } finally {
+      this.close();
+    }
   }
 
   @Override

--- a/src/main/java/io/github/janmalch/kino/security/AuthorizationFilter.java
+++ b/src/main/java/io/github/janmalch/kino/security/AuthorizationFilter.java
@@ -64,10 +64,13 @@ public class AuthorizationFilter implements ContainerRequestFilter {
   boolean userExists(Token token) {
     var repository = RepositoryFactory.createRepository(Account.class);
 
-    var query = new AccountByEmailSpec(token.getName(), repository);
-    var referredUser = repository.queryFirst(query);
-    repository.close();
-    return referredUser.isPresent();
+    try {
+      var query = new AccountByEmailSpec(token.getName(), repository);
+      var referredUser = repository.queryFirst(query);
+      return referredUser.isPresent();
+    } finally {
+      repository.close();
+    }
   }
 
   Token parseToken(String tokenString) {

--- a/src/main/java/io/github/janmalch/kino/security/TokenSecurityContext.java
+++ b/src/main/java/io/github/janmalch/kino/security/TokenSecurityContext.java
@@ -6,6 +6,7 @@ import io.github.janmalch.kino.repository.Repository;
 import io.github.janmalch.kino.repository.RepositoryFactory;
 import io.github.janmalch.kino.repository.specification.AccountByEmailSpec;
 import java.security.Principal;
+import java.util.Optional;
 import javax.ws.rs.core.SecurityContext;
 
 public class TokenSecurityContext implements SecurityContext {
@@ -24,8 +25,12 @@ public class TokenSecurityContext implements SecurityContext {
     Repository<Account> repository = RepositoryFactory.createRepository(Account.class);
 
     var query = new AccountByEmailSpec(_token.getName(), repository);
-    var referredUser = repository.queryFirst(query);
-    repository.close();
+    Optional<Account> referredUser;
+    try {
+      referredUser = repository.queryFirst(query);
+    } finally {
+      repository.close();
+    }
     if (referredUser.isEmpty()) {
       return null;
     }

